weld/llvm.rs

```diff
@@ -765,223 +781,279 @@ impl LlvmGenerator {
         Ok(())
     }
 
+    /*********************************************************************************************
+    //
+    //
+    //
+    // TODO(shoumik): Refactor code above this point!
+    //
+    //
+    //
+    ********************************************************************************************/
+
+    /*********************************************************************************************
+    //
+    // Utilities
+    //
+    *********************************************************************************************/
+
+    /// Returns the Type, LLVM type (as a string), and LLVM symbol name (as a string).
+    fn symbol_info<'a>(&mut self, func: &'a SirFunction, sym: &Symbol) -> WeldResult<(&'a Type, String, String)> {
+        let ty = func.symbol_type(sym)?;
+        let llvm_ty = self.llvm_type(ty)?;
+        let llvm_str = llvm_symbol(sym);
+        Ok((ty, llvm_ty, llvm_str))
+    }
```

Comment:
FYI I'm not sure that this is clearer than what we had before. I see a lot of `let (_, blah_ty, blah_sym) = symbol_info(...)`, which is pretty hard to parse -- someone unfamiliar with the code would not know what those are until they read the doc for `symbol_info`. It also seems weird to return 3 values and then throw one away. I think you can fix this in one of the following ways: either (1) return an Info object that has type, symbol, etc as fields, or (2) just use separate functions for each thing -- after all, I see no efficiency arguments in computing all three of these at once.